tests/storage/snapshot.go

```diff
@@ -834,6 +834,72 @@ var _ = Describe(SIG("VirtualMachineSnapshot Tests", func() {
 					}
 				})
 			})
+			It("should not include pending volumes", func() {
+				blankDv := func() *cdiv1.DataVolume {
+					return libdv.NewDataVolume(
+						libdv.WithNamespace(testsuite.GetTestNamespace(nil)),
+						libdv.WithBlankImageSource(),
+						libdv.WithStorage(
+							libdv.StorageWithStorageClass(snapshotStorageClass),
+							libdv.StorageWithVolumeSize(cd.BlankVolumeSize),
+						),
+						libdv.WithForceBindAnnotation(),
+					)
+				}
+
+				Expect(err).ToNot(HaveOccurred())
+				vmi := libvmifact.NewFedora(libnet.WithMasqueradeNetworking())
+				vmi.Namespace = testsuite.GetTestNamespace(nil)
+				vm = libvmi.NewVirtualMachine(vmi)
+				vm, vmi = createAndStartVM(vm)
+				libwait.WaitForSuccessfulVMIStart(vmi,
+					libwait.WithTimeout(300),
+				)
+				Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+				vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Adding a pending disk to the VM (non-hotplug)")
+				pendingDV := blankDv()
```

Comment:
Is "pending" a confusing term in this context, or do you actually enforce that the volume is pending somehow?